### ISSUE TITLE
[Bug]: Race condition when creating an application and volume

### ISSUE THREAD

**Issue by kieranbrown** [andrewbaxter/terraform-provider-fly#6]
### Description

I've noticed every so often upon initial app creation it tries to create subsequent resources too fast,  assumingly before the application exists on Fly.io, and errors with a 404 saying the app cannot be found.

This only occurs 50% of the time, if you keep applying/destroying you should be able to replicate.

I can work around this with a sleep of 1 second but I don't know if there can be anything added to the provider to make this more seamless. Perhaps either polling for existence after creation or even just adding retries with exponential backoff?

I've tried using `fly_app.this.name` `fly_app.this.id` and adding an explicit `depends_on`, even though `name/id` should implicitly add those dependencies, and in all scenarios I can still reproduce this bug.

This isn't a regression I can also replicate this on 0.1.11.

### Example error
```hcl
╷
│ Error: Failed to create volume
│ 
│   with fly_volume.this,
│   on main.tf line 21, in resource "fly_volume" "this":
│   21: resource "fly_volume" "this" {
│ 
│ got error response from POST https://api.machines.dev/v1/apps/random-app-name-race-condition-test/volumes: 404 Not Found
│ body:
```

### Test case

```hcl
terraform {
  required_providers {
    fly = {
      source = "andrewbaxter/fly"
      version = "0.1.13"
    }
  }
}

provider "fly" {
  fly_api_token = ""
}

resource "fly_app" "this" {
  name = "random-app-name-race-condition-test"
  org  = "personal"

  assign_shared_ip_address = false
}

resource "fly_volume" "this" {
  app    = fly_app.this.name
  name   = "data"
  region = "lhr"
  size   = 1
}
```

**Comment by andrewbaxter** [comment]
Sorry, I totally missed this issue...

I agree with you on all points, and thanks a lot for the thorough reports!  I think finding and resolving upstream consistency problems probably needs to be done on a case by case basis...

FWIW I couldn't reproduce the issue but I added a retry just in case, I don't think it should make things worse at least.

**Comment by andrewbaxter** [comment]
Released in `v0.0.18`
